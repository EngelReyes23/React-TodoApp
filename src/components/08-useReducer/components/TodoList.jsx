import React from "react";
import PropTypes from "prop-types";
import TodoListItems from "./TodoListItems";

const TodoList = ({ todos, handleDelete, handleToggle }) => {
	return (
		<ul className="list-group list-group-flush">
			{todos.map((todo, index) => (
				<TodoListItems
					key={todo.id}
					todo={todo}
					index={index}
					handleDelete={handleDelete}
					handleToggle={handleToggle}
				/>
			))}
		</ul>
	);
};

// Proptypes requeridos
TodoList.propTypes = {
	todos: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.number.isRequired,
			description: PropTypes.string.isRequired,
			done: PropTypes.bool.isRequired,
		}).isRequired
	).isRequired,
	handleDelete: PropTypes.func.isRequired,
	handleToggle: PropTypes.func.isRequired,
};

export default React.memo(TodoList).displayName = "TodoList";
