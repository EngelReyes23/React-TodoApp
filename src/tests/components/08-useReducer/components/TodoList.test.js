import { shallow } from "enzyme";
import TodoList from "../../../../components/08-useReducer/components/TodoList";
import { demoTodo } from "../../../fixtures/demoTodo";

describe("Pruebas en <TodoList />", () => {
	// Preparacion del entorno
	const handleToggle = jest.fn();
	const handleDelete = jest.fn();
	const wrapper = shallow(
		<TodoList
			todos={demoTodo}
			handleToggle={handleToggle}
			handleDelete={handleDelete}
		/>
	);

	// 1er Test - Renderizar correctamente el componente
	test("Debe de mostrar el componente correctamente", () => {
		expect(wrapper).toMatchSnapshot();
	});

	// 2do Test - Comprobar el numero de TODOs
	test("Debe de mostrar el numero de TODOs correctamente", () => {
		expect(wrapper.find("TodoListItems").length).toBe(demoTodo.length);
	});
});
