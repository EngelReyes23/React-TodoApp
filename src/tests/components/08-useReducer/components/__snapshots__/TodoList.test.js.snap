// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList /> Debe de mostrar el componente correctamente 1`] = `
<TodoList
  handleDelete={[MockFunction]}
  handleToggle={[MockFunction]}
  todos={
    Array [
      Object {
        "description": "Hola Mundo",
        "done": true,
        "id": 1,
      },
      Object {
        "description": "Engel Reyes",
        "done": false,
        "id": 2,
      },
      Object {
        "description": "Hola Mundo",
        "done": true,
        "id": 3,
      },
    ]
  }
/>
`;
